### PR TITLE
Add fsharp web templates

### PR DESCRIPTION
**Pull request by jmarolf**
**Customer scenario**

Customer attempts to create F# web projects from Visual Studio and finds no mechanism to do so.

**Bugs this fixes:** 

https://devdiv.visualstudio.com/DevDiv/_workitems/edit/549007

**Workarounds, if any**

Users can create project using the CLI

**Risk**

These templates are isolated from the rest of the project system (and Visual Studio) this risk is very low.

**Performance impact**

Templates do not touch any core perf scenarios.

**Is this a regression from a previous update?**
No

**New Template Experience**
image
image
image
  
**Note**: Depends on https://github.com/aspnet/templating/pull/254 being inserted

### CODE REVIEW

File: src/Templates/Microsoft.NetCore.FSharp.ProjectTemplates.Web/source.extension.vsixmanifest [hunk 1]
```diff
@@ -0,0 +1,24 @@
+<?xml version="1.0" encoding="utf-8"?>
+<PackageManifest Version="2.0.0" xmlns="http://schemas.microsoft.com/developer/vsx-schema/2011" xmlns:d="http://schemas.microsoft.com/developer/vsx-schema-design/2011">
+  <Metadata>
+    <Identity Id="Microsoft.NetCore.FSharp.ProjectTemplates.Web" Version="|%CurrentProject%;GetBuildVersion|" Language="en-US" Publisher="Microsoft" />
+    <DisplayName>F# .NETCore Project Templates Test</DisplayName>
+    <Description>F# project templates for ASP.NETCore.</Description>
+  </Metadata>
+  <Installation SystemComponent="true" Experimental="true">
+    <InstallationTarget Version="[15.0,]" Id="Microsoft.VisualStudio.Pro" />
+    <InstallationTarget Version="[15.0,]" Id="Microsoft.VisualStudio.VSWinDesktopExpress" />
```
Comment (KevinRansom) [inline src/Templates/Microsoft.NetCore.FSharp.ProjectTemplates.Web/source.extension.vsixmanifest]: Are the express skus still a thing?  I'm pretty sure we ripped out our support for them.

File: src/Templates/Microsoft.NetCore.FSharp.ProjectTemplates.Web/source.extension.vsixmanifest [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    <InstallationTarget Version="[15.0,]" Id="Microsoft.VisualStudio.VWDExpress" />
+    <InstallationTarget Version="[15.0,]" Id="Microsoft.VisualStudio.VSWinExpress" />
+  </Installation>
+  <Prerequisites>
+    <Prerequisite Id="Microsoft.VisualStudio.Component.CoreEditor" Version="[15.0,)" DisplayName="Visual Studio core editor" />
+    <Prerequisite Id="Microsoft.VisualStudio.ComponentGroup.WebToolsExtensions.TemplateEngine" Version="[15.0,)" DisplayName="ASP.NET templating engine" />
+  </Prerequisites>
+  <Dependencies>
+    <Dependency Id="Microsoft.Framework.NDP" DisplayName="Microsoft .NET Framework" d:Source="Manual" Version="[4.5,)" />
```
Comment (KevinRansom) [inline src/Templates/Microsoft.NetCore.FSharp.ProjectTemplates.Web/source.extension.vsixmanifest]: Is this actually a dependency?  After all they are dotnetcore templates?

File: src/Templates/Microsoft.NetCore.FSharp.ProjectTemplates.Web/source.extension.vsixmanifest (hunk 1 above)
Comment (jmarolf) [reply src/Templates/Microsoft.NetCore.FSharp.ProjectTemplates.Web/source.extension.vsixmanifest]: The express SKUs still ship.  I do not know if F# is installed in any of them.

File: src/Templates/Microsoft.NetCore.FSharp.ProjectTemplates.Web/Microsoft.NetCore.FSharp.ProjectTemplates.Web.csproj [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+<Project Sdk="Microsoft.NET.Sdk">
+  <PropertyGroup>
+    <!-- TODO: Disable deploying F# VSIXes for now: https://github.com/dotnet/project-system/issues/2575 -->
+    <DeployProjectOutput>false</DeployProjectOutput>
+  </PropertyGroup>
+  <ItemGroup>
+    <VSTemplate Include="Web.vstemplate">
+      <OutputSubPath>.NET Core</OutputSubPath>
+    </VSTemplate>
+        <VSTemplate Include="Web.vstemplate">
```
Comment (brettfo) [inline src/Templates/Microsoft.NetCore.FSharp.ProjectTemplates.Web/Microsoft.NetCore.FSharp.ProjectTemplates.Web.csproj]: Nit: the indention here makes me sad.

**Comment by jmarolf** [comment]
I am currently waiting for a validation build that contains all my changes.
[signed project-system build](https://devdiv.visualstudio.com/DevDiv/_build/index?buildId=1318225)
[insertion PR](https://devdiv.visualstudio.com/DevDiv/_git/VS/pullrequest/103494)
[insertion validation build](https://devdiv.visualstudio.com/DevDiv/_build/index?buildId=1322513)

**Comment by jmarolf** [comment]
[Template fixes](https://github.com/aspnet/templating/pull/281) have merged so this is safe to merge now. Just need to make sure these meet in the insertion PR

**Pull request merged**